lib/matplotlib/__init__.py

```diff
@@ -215,7 +217,78 @@ def _is_writable_dir(p):
     """
     return os.access(p, os.W_OK) and os.path.isdir(p)
 
+_verbose_msg = """--verbose-LEVEL and the Verbose class are deprecated.
+                  This functionality is now provided by the standard
+                  python logging library.  To get more (or less) logging
+                  output:
+                  `import logging`
+                  `logger=logging.getLogger('matplotlib')`
+                  `logger.set_level(logging.INFO)`
+                  """
 
+
+def _set_logger_verbose_level(level_str='silent', file_str='sys.stdout'):
+    """
+    Use a --verbose-LEVEL level to set the logging level:
+
+    """
+    levelmap = {'silent': logging.WARNING, 'helpful': logging.INFO,
+                'debug': logging.DEBUG, 'debug-annoying': logging.DEBUG,
+                'info': logging.INFO, 'warning': logging.warning}
+    # Check that current state of logger isn't already more verbose
+    # than the requested level.  If its more verbose, then leave more
+    # verbose.
+    if level_str in levelmap:
+        log = logging.getLogger(__name__)
+        newlev = levelmap[level_str]
+        oldlev = log.getEffectiveLevel()
+        if newlev < oldlev:
+            log.setLevel(newlev)
+            std = {
+                'sys.stdout': sys.stdout,
+                'sys.stderr': sys.stderr,
+            }
+            if file_str in std:
+                fileo = std[file_str]
+            else:
+                fileo = sys.stdout
+                try:
+                    fileo = open(file_str, 'w')
+                    # if this fails, we will just write to stdout
+                except IOError:
+                    warning.warn('could not open log file "{0}"'
+                                'for writing.\nCheck your '
```

Comment:
I tend not to prefer embedding newlines into this kind of messages because you never know what the width of the console is going to be anyways.  Not a big deal though.